core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -459,7 +459,23 @@ private[spark] class TaskSetManager(
           }
           // Serialize and return the task
           val startTime = clock.getTime()
+
           val serializedTask: ByteBuffer = try {
+            // We rely on the DAGScheduler to catch non-serializable closures and RDDs, so in here
+            // we assume the task can be serialized without exceptions.
+
+            // Check if serialization debugging is enabled
+            val debugSerialization: Boolean = sched.sc.getConf.
+              getBoolean("spark.serializer.debug", false)
+
+            if (debugSerialization) {
+              SerializationHelper.tryToSerialize(ser, task).fold (
```

Comment:
We should make sure this catches any exceptions thrown by the serialization utility itself and in that case just say that we couldn't produce debugging output.